src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -1838,4 +1895,64 @@ public with sharing class ADDR_Addresses_TEST {
         System.assertEquals(expected.MailingCountry__c, c.MailingCountry);
     }
 
+
+    /*******************************************************************************************************
+    * @description Builds SOQL for Contact sObject. Fields include standard Mailing fields as well as the one for State and Country Picklist enabled.
+    * @return String SOQL
+    */
+    private static String buildContactSoql() {
+        List<String> selectFields
+            = new List<String>{'Id', 'AccountId', 'is_Address_Override__c', 'Current_Address__c',
+                                'MailingStreet', 'MailingCity', 'MailingState', 'MailingCountry', 'MailingPostalCode',
+                                'MailingLatitude', 'MailingLongitude'};
+
+        if (ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            selectFields.add('MailingCountryCode');
+            selectFields.add('MailingStateCode');
+            selectFields.add('OtherCountryCode');
+            selectFields.add('OtherStateCode');
+        }
+
+        return new UTIL_Query()
+            .withSelectFields(selectFields)
+            .withFrom('Contact')
+            .build();
+    }
+
+
+    /*******************************************************************************************************
+    * @description Builds SOQL for Account sObject. Fields include standard Billing fields as well as the one for State and Country Picklist enabled.
+    * @return String SOQL
+    */
+    private static String buildAccountSoql() {
+        List<String> selectFields
+            = new List<String>{'Id', 'BillingStreet', 'BillingCity', 'BillingState',  'BillingCountry',
+                                'BillingPostalCode', 'BillingLatitude', 'BillingLongitude'};
+
+        if (ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            selectFields.add('BillingCountryCode');
+            selectFields.add('BillingStateCode');
+        }
+
+        return new UTIL_Query()
+            .withSelectFields(selectFields)
+            .withFrom('Account')
+            .build();
+    }
+
+
+    /*******************************************************************************************************
+    * @description Builds SOQL for Address sObject. Fields include standard Billing fields as well as the one for State and Country Picklist enabled.
+    * @return String SOQL
+    */
+    private static String buildAddressSoql() {
```

Comment:
Do we need a dynamic Query builder for the `Address__c` ? We already have a helper method to query Address in `getAddresses()`.